packages/bootstrap-vue-next/src/composables/useScrollspy.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  type ComponentPublicInstance,
+  computed,
+  getCurrentInstance,
+  type MaybeRefOrGetter,
+  nextTick,
+  onMounted,
+  readonly,
+  type Ref,
+  ref,
+  toValue,
+  watch,
+} from 'vue'
+import {useIntersectionObserver, useMutationObserver} from '@vueuse/core'
+import {getElement} from '../utils'
+
+type ScrollspyList = {
+  id: string | null
+  el: HTMLElement | null
+  visible: boolean
+  text: string | null
+}[]
+
+interface ScrollspyReturn {
+  current: Readonly<Ref<string | null>>
+  list: Readonly<Ref<ScrollspyList>>
+  content: Ref<HTMLElement | undefined>
+  target: Ref<HTMLElement | undefined>
+  scrollIntoView: (event: MouseEvent) => void
+  updateList: () => void
+  cleanup: () => void
+}
+
+interface ScrollspyOptions {
+  contentQuery: string
+  targetQuery: string
+  manual: boolean
+  root: string | ComponentPublicInstance | HTMLElement | null
+  rootMargin: string
+  threshold: number | number[]
+  watchChanges: boolean
+}
+
+export default (
+  content: MaybeRefOrGetter<string | ComponentPublicInstance | HTMLElement | null>,
+  target: MaybeRefOrGetter<string | ComponentPublicInstance | HTMLElement | null>,
+  options: Partial<ScrollspyOptions> = {}
+): ScrollspyReturn => {
+  // Cludge to get elements with dom query after mount
+  // because we can't trigger watch on mount with string values
+  const resolvedContent = ref(getElement(content))
+  const resolvedTarget = ref(getElement(target))
+  const updateElements = () => {
+    resolvedContent.value = getElement(content)
+    resolvedTarget.value = getElement(target)
+  }
+  watch([() => toValue(content), () => toValue(target)], () => {
+    updateElements()
+    updateList()
+  })
```

Comment:
```suggestion
  const resolvedContent = readonly(toRef(content))  
  const resolvedTarget = readonly(toRef(target))
  
  const resContent = computed(() => getElement(resolvedContent.value))
  const resTarget = computed(() => getElement(resolvedTarget.value))
  
  watch([resTarget, resContent], () => {
    updateList()
  })
```

Or
```suggestion
  const resolvedContent = computed(() => getElement(toValue(content)))  
  const resolvedTarget = computed(() => getElement(toValue(target)))
  
  watch([resolvedTarget, resolvedContent], () => {
    updateList()
  })
```

Subsequently usage lower can use typical `.value` to get values, such as `if (typeof toValue(content)` => `if(typeof resolvedContent.value...`

Reply:
that doesn't work if the input is string. That is the reason for that cludge. Or do we want to remove the functionality of us doing the querying and only accept input as elements or refs on every component and directive?

Reply:
There should be no reason why it doesn't work. In both cases the string and HTML element should be gathered

Reply:
The string doesn't update when the dom is "mounted" and the composable will have no element in the variable. We need to update it in onMounted if it's a string.

Reply:
Depending on what you'd want, you can either exclude the _readonly_ and sync the two refs (I don't think you can do that with a maybe ref or getter, as you can't sync a ref to a function, just doesn't make sense) or you can have it one way sync with a watcher, and you'll need to declare a ref. The second is what you're doing here, but you can write it more succinct by taking some of the recommended changes -- namely adding the toRef

Reply:
how does toRef help? How do I make it update it's return value?